holidays/countries/chile.py

```diff
@@ -64,127 +68,174 @@ class Chile(HolidayBase):
         self[date(year, JAN, 1)] = "Año Nuevo [New Year's Day]"
         # Day after, if it's a Sunday (Law 20.983)
-        if year > 2016 and date(year, JAN, 1).weekday() == SUN:
+        if year >= 2017 and date(year, JAN, 1).weekday() == SUN:
             self[date(year, JAN, 2)] = "Fiestas Patrias [Holiday]"
 
         # Holy Week (Law 2.977)
-        name_fri = "Semana Santa (Viernes Santo) [Good Friday)]"
-        name_sat = "Semana Santa (Sábado Santo) [Good Saturday)]"
-        name_easter = "Día de Pascuas [Easter Day]"
-
         easter_date = easter(year)
-        self[easter_date + rd(days=-2)] = name_fri
-        self[easter_date + rd(days=-1)] = name_sat
-        self[easter_date] = name_easter
+        self[
+            easter_date + rd(days=-2)
+        ] = "Semana Santa (Viernes Santo) [Good Friday)]"
+        self[
+            easter_date + rd(days=-1)
+        ] = "Semana Santa (Sábado Santo) [Good Saturday)]"
+
+        # Ascension
+        if year <= 1967:
+            self[
+                easter_date + rd(days=+39)
+            ] = "Ascensión del Señor [Ascension of Jesus]"
+
+        # Corpus Christi
+        if year <= 1967 or 1987 <= year <= 2006:
+            # Law 19.668
+            if year <= 1999:
+                dt = easter_date + rd(days=+60)
+            else:
+                dt = easter_date + rd(days=+57)
+            self[dt] = "Corpus Christi [Corpus Christi]"
 
-        # Labor Day (Law 2.200, renamed with Law 18.018)
-        name = "Día Nacional del Trabajo [Labour Day]"
-        self[date(year, MAY, 1)] = name
+        # Labour Day (Law 2.200, renamed with Law 18.018)
+        if year >= 1932:
+            self[date(year, MAY, 1)] = "Día Nacional del Trabajo [Labour Day]"
 
         # Naval Glories Day (Law 2.977)
-        name = "Día de las Glorias Navales [Navy Day]"
-        self[date(year, MAY, 21)] = name
-
-        name = "Día Nacional de los Pueblos Indígenas"
-        if year == 2021:
-            self[date(year, JUN, 21)] = name
-        if year > 2021:
-            self[date(year, JUN, 19)] = name
-
-        # Saint Peter and Saint Paul (Law 18.432)
-        name = "San Pedro y San Pablo [Saint Peter and Saint Paul]"
-        if year < 2020:
-            self[date(year, JUN, 29)] = name
-        else:
-            # floating Monday holiday (Law 19.668)
-            if date(year, JUN, 29).weekday() <= THU:
-                self[
-                    date(year, JUN, 29)
-                    + rd(date(year, JUN, 29), weekday=MO(-1))
-                ] = name
-            elif date(year, JUN, 29).weekday() == FRI:
-                self[date(year, JUN, 29) + rd(weekday=MO)] = name
-            else:
-                self[date(year, JUN, 29)] = name
+        self[date(year, MAY, 21)] = "Día de las Glorias Navales [Navy Day]"
+
+        # National Day of Indigenous Peoples
+        if year >= 2021:
+            dt = date(year, JUN, 21)
+            if year >= 2022 and (year - 2022) % 4 > 1:
```

Comment:
Could you add a comment explaining what's going on here?

Reply:
It's summer solstice :) For this holiday may be better use code from #828

Reply:
Just updated my code to be fully compliant for whole 21st century.